kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteFileProcessor.java

```diff
@@ -0,0 +1,315 @@
+package io.kaoto.backend.metadata.parser.step.camelRoute;
+
+import io.kaoto.backend.metadata.parser.JsonProcessFile;
+import io.kaoto.backend.model.parameter.BooleanParameter;
+import io.kaoto.backend.model.parameter.ObjectParameter;
+import io.kaoto.backend.model.parameter.NumberParameter;
+import io.kaoto.backend.model.parameter.StringParameter;
+import io.kaoto.backend.model.parameter.Parameter;
+import io.kaoto.backend.model.step.Step;
+import org.jboss.logging.Logger;
+
+import javax.json.Json;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CamelRouteFileProcessor extends JsonProcessFile<Step> {
+
+    private static final String SOURCE_TYPE = "source";
+    private static final String SINK_TYPE = "sink";
+    private static final String ACTION_TYPE = "action";
+
+    record JsonCamelObject(
+        String id,
+        String kind,
+        String icon,
+
+        String title,
+        String description,
+        String group,
+        String type,
+        List<Parameter> parameters,
+        List<String> required
+    ) { }
+    private final Logger log = Logger.getLogger(CamelRouteFileProcessor.class);
+
+    @Override
+    protected List<Step> parseFile(final File f) {
+        try (FileReader fr = new FileReader(f)) {
+
+            JsonReader reader = Json.createReader(fr);
+            JsonObject json = reader.readObject();
+            Step step = convertToStep(json);
+
+            List<Step> duplicatedStepsToTypes = List.of();
+
+            if (step != null) {
+                duplicatedStepsToTypes = duplicateStepToOtherTypes(step);
+            }
+
+            return duplicatedStepsToTypes;
```

Comment:
The above approach is preferred over a if else because of ciclomatic complexity.

Think about this, we know that the default value is an empty list. We have a single point of return. We have less identations and scopes. The Else really is not adding "much".

Plus the else is harder to maintain if we have to do extra steps or stuff in the future.